pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -42,23 +40,41 @@ const wildcardDomainPrefix = "*."
 
 // BuildHTTPRoutes produces a list of routes for the proxy
 func (configgen *ConfigGeneratorImpl) BuildHTTPRoutes(env *model.Environment, node *model.Proxy, push *model.PushContext,
-	routeName string) *xdsapi.RouteConfiguration {
-	var rc *xdsapi.RouteConfiguration
+	routeNames []string) []*xdsapi.RouteConfiguration {
+	routeConfigurations := make([]*xdsapi.RouteConfiguration, 0)
+
 	switch node.Type {
 	case model.SidecarProxy:
-		rc = configgen.buildSidecarOutboundHTTPRouteConfig(env, node, push, routeName)
-		if rc != nil {
-			rc = envoyfilter.ApplyRouteConfigurationPatches(networking.EnvoyFilter_SIDECAR_OUTBOUND, node, push, rc)
+		vHostCache := make(map[int][]*route.VirtualHost)
+		for _, routeName := range routeNames {
+			rc := configgen.buildSidecarOutboundHTTPRouteConfig(env, node, push, routeName, vHostCache)
```

Comment:
I am not sure how can this happen?

I deployed bookinfo and dumped the route and only  `"inbound|9080|http|ratings.default.svc.cluster.local"` repeated 4 times, and they are all from listeners filters not dynamic routes.

Reply:
I noticed the route includes `[15004 15010 15014 20001 3000 80 8060 8080 9080 9090 9091 9411 9901 istio-ingressgateway.istio-system.svc.cluster.local:15020 istio-telemetry.istio-system.svc.cluster.local:42422 jaeger-collector.istio-system.svc.cluster.local:14267 jaeger-collector.istio-system.svc.cluster.local:14268 jaeger-query.istio-system.svc.cluster.local:16686 productpage.default.svc.cluster.local:9080]` when i rename the port name of productpage, 2 `9080` requires calculate

Reply:
It works for outbound route `9080` and `productpage.default.svc.cluster.local:9080`